client/src/components/QuestionBox.tsx

```diff
@@ -0,0 +1,94 @@
+import { useState } from "react";
+import { useQueryClient } from "@tanstack/react-query";
+import AnswersAndVotes from "./AnswersAndVoters";
+import Button from "./Button";
+import * as adminQuickRound from "../newtork/adminQuickRound";
+import EditQuestionForm from "./EditQuestionForm";
+import { Modal, ModalHeader, ModalBody } from "./Modal";
+
+interface Answer {
+  [key: string]: number;
+}
+
+interface QuestionProps {
+  id: number;
+  question: string;
+  answers: Answer;
+  createdAt: string;
+  updatedAt: string;
+}
+
+function convertDate(date: string) {
+  return new Date(date).toLocaleString();
+}
+
+export default function QuestionBox({ question }: { question: QuestionProps }) {
+  const queryClient = useQueryClient();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  async function handleDelete(id: number) {
+    try {
+      await adminQuickRound.deleteQuestion(id);
+      queryClient.invalidateQueries({ queryKey: ["questions"] });
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  function handleEdit() {
+    setIsModalOpen(true);
+  }
+
+  interface UpdatedDataProps {
+    question: string;
+    answers: { [key: string]: number };
+  }
+
+  const handleEditSubmit = async (data: UpdatedDataProps) => {
+    try {
+      await adminQuickRound.updateQuestion(question.id, data);
+      queryClient.invalidateQueries({ queryKey: ["questions"] });
```

Comment:
i think here you need an `await` keyword

Reply:
`queryClient.invalidateQueries({ queryKey: ["questions"] });` is a non-blocking operation, and the UI and application logic don't need to wait for this process to complete.